### ISSUE TITLE
LatLonShapeDocValuesQueryTests.testIndexMultiShapes

### ISSUE THREAD

**Issue by romseygeek** [elastic/elasticsearch#85177]
### CI Link

n/a

### Repro line

./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.index.mapper.LatLonShapeDocValuesQueryTests.testIndexMultiShapes" -Dtests.seed=368A99A4FD18272B -Dtests.locale=lv -Dtests.timezone=Africa/Niamey -Druntime.java=17

### Does it reproduce?

Yes

### Applicable branches

7.17

### Failure excerpt

```
java.lang.AssertionError: expected:<0> but was:<1>
        at __randomizedtesting.SeedInfo.seed([368A99A4FD18272B:EABCA164BFB771B4]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at org.elasticsearch.xpack.spatial.index.mapper.LatLonShapeDocValuesQueryTests.assertQueries(LatLonShapeDocValuesQueryTests.java:148)
        at org.elasticsearch.xpack.spatial.index.mapper.LatLonShapeDocValuesQueryTests.testIndexMultiShapes(LatLonShapeDocValuesQueryTests.java:141)
```

**Comment by romseygeek** [comment]
This failed locally for me on a backport branch, and further testing showed that it is reproducible directly on the 7.17 branch.

**Comment by iverase** [comment]
This issue was due to a bug in Lucene so it trigger the following fix: https://issues.apache.org/jira/browse/LUCENE-10514

This fix is part of Lucene 9.2, but still can happen in Elasticsearch 7.17.x. I don't think it is worthy trying to backport as the error is exotic and should not happen in real usage. Maybe we should modify the test in 7.17 so all shapes from the multishape are disjoint?

**Comment by iverase** [comment]
fixed on https://github.com/elastic/elasticsearch/pull/87362
